### DISCUSSION TITLE
Implement multiple group chats flow

### DISCUSSION THREAD

**Discussion by IgorKhomenko** [uNetworking/uWebSockets.js#725] [General]
Hi everyone

I have a use case with a standard chat app:

- there are many users 
- there are also many group chats 
- each user can be a member of let's say 300 group chats

From what I got, we can use pub/sub for the group chats implementation. 

The question is the following: 
Each user should be able to receive messages from any group (where a user is a member) in a real time. When a user connects to server, should we loop against all groups where a user is member and subscribe to each?  

E.g.

```
const app = uWS.App({

}).ws('/*', {

  /* Handlers */
  open: (ws) => {
    const userGroups = getUserGroups(ws);
    for (let groupId of userGroups) {
        ws.subscribe(groupId);
    }
  },
  message: (ws, message, isBinary) => {

  },
  drain: (ws) => {

  },
  close: (ws, code, message) => {

  }
}).any('/*', (res, req) => {

}).listen(port, (token) => {

});
```

Is there any better way implement it?

**Comment by e3dio** [comment]
Seems fine, not sure what you think is wrong? ``ws.subscribe(`group/${id}`)``

**Reply by IgorKhomenko** [reply]
@e3dio I think this 'for loop' for 300 iterations may cause some performance issues

**Reply by e3dio** [reply]
Completely insignificant and miniscule, you are even doing correctly with 'for of' for an array. Usually you don't send every message for all 300 groups unless you have unique app, you send messages for group user is currently viewing, you would have separate subscription for notifications that a group got a message, to minimize the amount of data sent to user which is more important

**Reply by IgorKhomenko** [reply]
@e3dio ok this does make a lot of sense. Thanks for your help 👍
